marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/steps/MappingStepController.java

```diff
@@ -83,6 +84,20 @@
         return new ResponseEntity<>(body, headers, HttpStatus.OK);
     }
 
+    @RequestMapping(value = "/{stepName}/newDoc", method = RequestMethod.GET)
+    @ApiOperation(value = "Returns an XML or JSON source document as a string of JSON")
+    @Secured("ROLE_readMapping")
+    public ResponseEntity<JsonNode> getNewDocument(@PathVariable String stepName, @PathVariable String docUri) {
+        // Step defines which database to pull the identified doc from.
+        final JsonNode stepNode = newService().getStep(STEP_DEFINITION_TYPE, stepName);
+        final String databaseName = stepNode.get("sourceDatabase").asText();
+        // In the hub context, we're presuming staging or final.
+        final DatabaseClient databaseClient = databaseName.toLowerCase().equals(getHubClient().getDbName(DatabaseKind.STAGING).toLowerCase()) ?
+            getHubClient().getStagingClient() :
+            getHubClient().getFinalClient();
+        return ResponseEntity.ok(MappingService.on(databaseClient).getNewDocument(stepName, docUri));
```

Comment:
The other endpoints herein use `StepService`.  Is it okay that this method uses `MappingService`?

Reply:
That's finr